### PR TITLE
Some tweaks and cleanup for GASMAN and Boehm

### PR DESCRIPTION
**Pull request by fingolfin**
This is a subset of PR #1471. The latter currently conflicts with `master`, and by moving parts of it to a separate PR, I am hoping they can be merged quicker.

### CODE REVIEW

File: src/gasman.c [hunk 1]
```diff
@@ -324,7 +324,7 @@ void CHANGED_BAG_IMPL(Bag bag) {
 #define SpaceBetweenPointers(a,b) (((UInt)((UInt)(a) - (UInt)(b)))/sizeof(Bag))
 
 #define SizeMptrsArea SpaceBetweenPointers(OldBags, MptrBags)
-#define SizeOldBagsArea SpaceBetweenPointers(YoungBags,OldBags)
+#define SizeOldBagsArea SpaceBetweenPointers(YoungBags, OldBags)
 #define SizeYoungBagsArea SpaceBetweenPointers(AllocBags, YoungBags)
 #define SizeAllocationArea SpaceBetweenPointers(StopBags, AllocBags)
 #define SizeUnavailableArea SpaceBetweenPointers(EndBags, StopBags)
```
Comment (fingolfin) [inline src/gasman.c]: BTW, I don't quite understand the name `SizeUnavailableArea`.  Isn't this rather *available* space? Resp. free, unused, ... ? @stevelinton @ChrisJefferson what am I missing?

Comment (fingolfin) [inline src/gasman.c]: OK, what I missed was in particular the separate allocation area, and also the big comment around line 266.

But I still don't understand what this "unavailable" area is needed for...

Comment (stevelinton) [inline src/gasman.c]: This relates to the idea of limiting the size of the young generation to fit in cache. StopBags denotes the point at which a GC of the young generation should be triggered, EndBags represents the limit of memory obtained from the OS so the space between is voluntarily unavailable for allocations. Not a great name, and in fact that feature never worked well.

Comment (stevelinton) [inline src/gasman.c]: Comments crossed. It's really a reserve area. We will try a GC of the young generation before using it, but we haven't given it back to the OS.

File: src/gasman.c [hunk 2]
```diff
@@ -1805,82 +1799,6 @@ UInt CollectBags (
 
     /* * * * * * * * * * * * * * * sweep phase * * * * * * * * * * * * * * */
 
-#if 0
-    /* call freeing function for all dead bags                             */
-    if ( NrTabFreeFuncBags ) {
```
Comment (fingolfin) [inline src/gasman.c]: @stevelinton Do you by chance remember what this commented out code was good for?

File: src/gasman.c (hunk 1 above)
Comment (fingolfin) [inline src/gasman.c]: So pehaps "reserved area" would be a better / less confusing name?

File: src/gasman.c [hunk 3]
```diff
@@ -2116,25 +2036,25 @@ UInt CollectBags (
         AllocSizeBags = 7*(Endbags - StopBags)/8; */
```
Comment (fingolfin) [inline src/gasman.c]: TODO: remove dead code above

File: src/gasman.c (hunk 2 above)
Comment (stevelinton) [inline src/gasman.c]: I have no memory of why, but looking at the code it seems there was a version which called the C level finalisers  in a separate pass before the sweep pass, and that this was later integrated with the sweep pass. At a quick glancce, no C level finalisers are ever set anyway, so, unless it's used in libsing or something, that whole feature could be dropped.

Comment (fingolfin) [inline src/gasman.c]: They are used in libsing and NormalizInterface and soon more. And also HPC-GAP (though of course we don't use GASMAN there)

File: src/gasman.c (hunk 1 above)
Comment (stevelinton) [inline src/gasman.c]: Yes, unless we want to take this opportunity to remove relics of this feature entirely.

**Comment by ChrisJefferson** [comment]
2nd commit title - "when COUNT_BAGS is" ... what? :)

File: src/gasman.c (hunk 1 above)
Comment (fingolfin) [inline src/gasman.c]: OK, so I'll also look into whether we can get rid of `StopBags` completely ;-)

Comment (fingolfin) [inline src/gasman.c]: OK, looking at `StopBags` more closely was actually quite illuminating (thanks for stumping my nose on it, @stevelinton !). I discovered this line of code (around line 1958):
```
    /* temporarily store in 'StopBags' where this allocation takes us      */
    StopBags = AllocBags + WORDS_BAG(sizeof(BagHeader)+size);
```
So `StopBags` is indeed not what it "normally" is, and the `SpaceBetweenPointers` are misleading. I have therefore dropped those from this PR, and will instead look into removing StopGaps in another PR (I'll investigate it, anyway)

**Pull request merged**